src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.cpp

```diff
@@ -75,116 +84,221 @@ ShenandoahHeuristics::~ShenandoahHeuristics() {
 void ShenandoahHeuristics::choose_collection_set(ShenandoahCollectionSet* collection_set) {
   assert(collection_set->count() == 0, "Must be empty");
 
-  ShenandoahHeap* heap = ShenandoahHeap::heap();
+  if (_generation->generation_mode() == OLD) {
+    // Old-gen doesn't actually choose a collection set to be evacuated by its own gang of worker tasks.
+    // Instead, it computes the set of regions to be evacuated by subsequent young-gen evacuation passes.
+    prepare_for_old_collections();
+  } else {
+    ShenandoahHeap* heap = ShenandoahHeap::heap();
```

Comment:
I'd factor out the whole else block here to call prepare_for_young_collection() or whatever is fitting there.

Reply:
I've refactored this code.  This comment is no longer relevant.